### PR TITLE
Add property map to ActionInterface to ease JSON schema generation for plugin actions

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

File: brayns/common/types.h [hunk 1]
```diff
@@ -383,6 +383,15 @@ enum class DataType
 
 class PropertyMap;
 class PropertyObject;
+
+/** Description for RPC with one parameter. */
+struct RpcDescription
+{
+    std::string methodName;
+    std::string methodDescription;
+    std::string paramName{};
```
Comment (karjonas) [inline brayns/common/types.h]: Remove {}.

File: brayns/common/types.h [hunk 2]
```diff
@@ -383,6 +383,15 @@ enum class DataType
...
+    std::string paramDescription{};
```
Comment (karjonas) [inline brayns/common/types.h]: Remove {}.

File: plugins/RocketsPlugin/RocketsPlugin.cpp [hunk 3]
```diff
@@ -868,14 +874,13 @@ class RocketsPlugin::Impl
...
-        _handleTask<SnapshotParams, ImageGenerator::ImageBase64>(
-            METHOD_SNAPSHOT, doc, func);
+        _handleTask<SnapshotParams, ImageGenerator::ImageBase64>(desc, func);
     }
 
     void _handleStreamTo()
     {
-        RpcDocumentation doc{"Stream to a displawall", "param",
-                             "Stream parameters"};
+        const RpcDescription desc{METHOD_STREAM_TO, "Stream to a displawall",
```
Comment (karjonas) [inline plugins/RocketsPlugin/RocketsPlugin.cpp]: displa(y)wall :)

File: plugins/RocketsPlugin/RocketsPlugin.cpp [hunk 4]
```diff
@@ -1142,6 +1147,77 @@ void RocketsPlugin::postSceneLoading()
...
+        name = desc.methodName, input, action, engine = _impl->_engine
+    ](const auto& request) {
+        PropertyMap params = input;
+        if (::from_json(params, request.message))
+        {
+            engine->triggerRender();
+            return Response{to_json(action(params))};
+        }
+        return Response{
+            Response::Error{"from_json for " + name + " failed", -42}};
```
Comment (karjonas) [inline plugins/RocketsPlugin/RocketsPlugin.cpp]: If -42 is an error code than it should have a proper define somewhere.

**Comment by tribal-tec** [comment]
ping

**Pull request merged**